### PR TITLE
feat: add SetContainsElements API

### PR DESCRIPTION
**Pull request by cprice404**

### CODE REVIEW

File: tests/Cache/CacheClientTest.php [hunk 1]
```diff
@@ -2799,6 +2858,129 @@ public function testSetAddElementsSetFetch_RefreshTtl()
...
+
+        $response = $this->client->setAddElements($this->TEST_CACHE_NAME, $setName, ["foo", "bar", "baz", "bam", "beep", "boop"], CollectionTtl::fromCacheTtl()->withNoRefreshTtlOnUpdates());
+        $this->assertNull($response->asError());
+        $this->assertNotNull($response->asSuccess(), "Expected a success but got: $response");
+
+        $response = $this->client->setContainsElements($this->TEST_CACHE_NAME, $setName, $elements);
+        $this->assertNull($response->asError());
+        $this->assertNotNull($response->asHit(), "Expected a hit but got: $response");
+
+        $expectedExists = [true, true, true];
```
Comment (rishtigupta) [inline tests/Cache/CacheClientTest.php:2897]: Shouldn't this be: [true, true, true, false, false, false]?

File: tests/Cache/CacheClientTest.php [hunk 2]
```diff
@@ -2799,6 +2858,129 @@ public function testSetAddElementsSetFetch_RefreshTtl()
...
+
+        $response = $this->client->setAddElements($this->TEST_CACHE_NAME, $setName, ["bar", "bam", "beep", "boop"], CollectionTtl::fromCacheTtl()->withNoRefreshTtlOnUpdates());
+        $this->assertNull($response->asError());
+        $this->assertNotNull($response->asSuccess(), "Expected a success but got: $response");
+
+        $response = $this->client->setContainsElements($this->TEST_CACHE_NAME, $setName, $elements);
+        $this->assertNull($response->asError());
+        $this->assertNotNull($response->asHit(), "Expected a hit but got: $response");
+
+        $expectedExists = [false, true, false];
```
Comment (rishtigupta) [inline tests/Cache/CacheClientTest.php:2921]: same comment as above.

File: src/Cache/CacheOperationTypes/CacheOperationTypes.php [hunk 3]
```diff
@@ -2868,6 +2869,119 @@ class SetAddElementsError extends SetAddElementsResponse
...
+ *
+ * * SetContainsElementsHit
+ * * SetContainsElementsMiss
+ * * SetContainsElementsError
+ *
+ * Pattern matching can be used to operate on the appropriate subtype.
+ * For example:
+ * <code>
+ * if ($success = $response->asHit()) {
+ *     // a dictionary indicating which elements were found in the set
```
[the remark this answers is not in the document]
Comment (cprice404) [reply src/Cache/CacheOperationTypes/CacheOperationTypes.php]: i changed the comment to try to make it more clear.

i'm not sure I know what you mean when you say "the method names do not match the response type"? can you suggest what you'd recommend that I change?

I chose `containsElements` for the values function based on the fact that that is the name we used in the JS SDK.

File: tests/Cache/CacheClientTest.php (hunk 1 above)
Comment (cprice404) [reply tests/Cache/CacheClientTest.php:2897]: no, my request only passed in `$elements`, which has 3 items in it.

File: src/Cache/CacheOperationTypes/CacheOperationTypes.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (cprice404) [reply src/Cache/CacheOperationTypes/CacheOperationTypes.php]: ah i see what you mean now, fixed in https://github.com/momentohq/client-sdk-php/pull/209/commits/ccdfa0017ae4cdb663c11b8cad5e9dce898dc5e1

**Review by rishtigupta — APPROVED** [review]
:shipit:

**Pull request merged**